src/Nethermind/Nethermind.Facade/Eth/RpcTransaction/AuthorizationListForRpc.cs

```diff
@@ -0,0 +1,86 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System.Collections.Generic;
+using System.Linq;
+using Nethermind.Core;
+using Nethermind.Int256;
+using System.Text.Json.Serialization;
+using System.Text.Json;
+using System;
+using Nethermind.Core.Crypto;
+
+namespace Nethermind.Facade.Eth.RpcTransaction;
+
+[JsonConverter(typeof(JsonConverter))]
+public class AuthorizationListForRpc
+{
+    private readonly IEnumerable<Tuple> _tuples;
+
+    [JsonConstructor]
+    public AuthorizationListForRpc() { }
+
+    private AuthorizationListForRpc(IEnumerable<Tuple> tuples)
+    {
+        _tuples = tuples;
+    }
+
+    private class Tuple
+    {
+        public UInt256 ChainId { get; set; }
+        public ulong Nonce { get; set; }
+        public Address Address { get; set; }
+
+        // TODO: Are these fields actually nullable
+        public UInt256? YParity { get; set; }
+        public UInt256? S { get; set; }
+        public UInt256? R { get; set; }
+
+        [JsonConstructor]
+        public Tuple() { }
+
+        public Tuple(UInt256 chainId, ulong nonce, Address address, UInt256? yParity, UInt256? s, UInt256? r)
+        {
+            ChainId = chainId;
+            Nonce = nonce;
+            Address = address;
+            YParity = yParity;
+            S = s;
+            R = r;
+        }
+    }
+
+    public static AuthorizationListForRpc FromAuthorizationList(IEnumerable<AuthorizationTuple>? authorizationList) =>
+        authorizationList is null
+        ? new AuthorizationListForRpc([])
+        : new AuthorizationListForRpc(authorizationList.Select(tuple =>
+            new Tuple(tuple.ChainId,
+                    tuple.Nonce,
+                    tuple.CodeAddress,
+                    tuple.AuthoritySignature.RecoveryId,
+                    new UInt256(tuple.AuthoritySignature.S),
+                    new UInt256(tuple.AuthoritySignature.R))));
+
+    public AuthorizationTuple[] ToAuthorizationList() => _tuples
+        .Select(tuple => new AuthorizationTuple(
+            (ulong)tuple.ChainId,
+            tuple.Address,
+            tuple.Nonce,
+            // TODO: This might be wrong
+            new Signature(tuple.R ?? 0, tuple.S ?? 0, (ulong)(tuple.YParity ?? 0)))
```

Comment:
@ak88 In the 7702 PR (https://github.com/NethermindEth/nethermind/pull/7459) there was no code that took a `TransactionForRpc` with a valid `AuthorizationList` and constructed a `Transaction` (we were never copying the `AuthorizationList` property).

This PR adds such behavior but I'm not sure if this exact line is correct.

Reply:
We can drop the nullability

Reply:
Nullability dropped, we'll revisit as needed